### PR TITLE
Add seq and historical RPC endpoints

### PR DESCRIPTION
**Pull request by boyuan-chen**
Add historical and sequencer RPC endpoints.

- The eth_sendRawTransaction is redirected to the sequencer node for the replica
- The legacy methods are redirected to the historical RPC endpoint.

The following methods are sent the historical RPC endpoint:
* `eth_call`
* `debug_traceTransaction`
* `eth_estimateGas`
* `eth_createAccessList`
* `eth_getStorageAt`
* `eth_getCode`
* `eth_getTransactionCount`
* `eth_getBalanace`

### CODE REVIEW

**Comment by boyuan-chen** [comment]
In `core/state_transistion.go`, the `if..else..` uses `IsBedrock` as the determination. Do we want to switch to `IsOptimismBedrock`? The difference between `IsBedrock` and `IsOptimismBedrock` is that `IsOptimismBedrock` checks whether the chain configuration is for the optimism. `IsBedrock` only checks whether the block number is larger the `bedrockBlockNumber`. The default value of `bedrockBlockNumber` is 0, so `IsBedrock` is always active.

This change might break tests.

```
	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && st.evm.ChainRules().IsBedrock {
		l1Cost = st.evm.Context().L1CostFunc(st.evm.Context().BlockNumber, st.msg)
		if l1Cost != nil {
			if _, overflow = mgval.AddOverflow(mgval, l1Cost); overflow {
				return fmt.Errorf("%w: address %v", ErrInsufficientFunds, st.msg.From().Hex())
			}
		}
	}
```

File: cmd/rpcdaemon/commands/tracing.go [hunk 1]
```diff
@@ -200,6 +201,21 @@ func (api *PrivateDebugAPIImpl) TraceTransaction(ctx context.Context, hash commo
...
+		if api.historicalRPCService != nil {
+			treeResult := &GethTrace{}
+			if err := api.historicalRPCService.CallContext(ctx, treeResult, "debug_traceTransaction", hash, config); err != nil {
+				return fmt.Errorf("historical backend failed: %w", err)
+			}
+			result, err := json.Marshal(treeResult)
+			if err != nil {
+				return err
+			}
+			stream.WriteRaw(string(result))
```
Comment (jyellick) [inline cmd/rpcdaemon/commands/tracing.go]: I'd assume we want to return `nil` or capture an error here? (As written, I think we always fall through to return `rpc.ErrNoHistoricalFallback`)

**Review by jyellick — COMMENTED** [review]
Overall LGTM, only one small thing.

With respect to the `IsBedrock` vs. `IsOptimismBedrock`, the if statement already checks for `optimismConfig != nil` from the chain config, so I actually think it's equivalent?

**Review by mmontour1306 — APPROVED** [review]

File: cmd/rpcdaemon/commands/tracing.go (hunk 1 above)
Comment (boyuan-chen) [reply cmd/rpcdaemon/commands/tracing.go]: Good catch! Fixed

**Comment by boyuan-chen** [comment]
The unit tests are added.

**Comment by boyuan-chen** [comment]
> Overall LGTM, only one small thing.
> 
> With respect to the `IsBedrock` vs. `IsOptimismBedrock`, the if statement already checks for `optimismConfig != nil` from the chain config, so I actually think it's equivalent?

Yes!

**Review by jyellick — APPROVED** [review]

**Pull request merged**